src/client/util/http.h

```diff
@@ -172,15 +176,19 @@ class HTTP : public HTTPStorage {
 
   /// @brief Get initialized URI
   /// @return cpp-netlib URI object
-  boost::network::uri::uri GetURI() {
+  boost::network::uri::uri GetURI() const
+  {
     return m_URI;
   }
 
-  /// @brief Tests if TLD is I2P and set default ports for in-net downloading
+  /// @brief Tests if TLD is I2P
   /// @return True if TLD is .i2p
+  bool HostIsI2P() const;
+
+  /// @brief Set default ports for in-net downloading
   /// @notes The default port is 80, for HTTPS it is 443
   /// @notes Removing this will require refactoring stream implementation
-  bool HostIsI2P();
+  void CompleteURI();
```

Comment:
>CompleteURI();

I like how you want to decouple host type check from the amending of the URI. I must point out though that the only use-case we have for amending the URI is for in-net downloading and the only reason we need to amend the URI for in-net downloading is because the cpp-netlib refactor isn't finished. Also, adding this function implies that there is a use-case to amend the URI elsewhere - which there isn't.

If we want to decouple this, maybe we should also change the logic test for host type (e.g., `if (HostIsI2P()) { AmendURI(); return DownloadViaI2P(); }`. Ultimately, I think the best solution is for the refactor to be finished but I think that would be fine for now.

Did you have any other thoughts on this area?